File: Oregon 9-1-1/Oregon 9-1-1.Android.NativeActivity/Unit.cpp
#include "Oregon911.h"

Unit::Unit(unsigned int _id, std::string _name, std::string _agency, std::string _station, std::string _status[MAX_STATUS]) {

}

Unit::Unit() {
	id = 0;
}

Unit::~Unit()
{
}

unsigned int Unit::getid()
{
	return id;
}

std::string Unit::getName()
{
	return name;
}

std::string Unit::getAgency()
{
	return agency;
}

std::string Unit::getStation()
{
	return station;
}

void Unit::setName(const std::string& _name)
{
	name = _name;
}

void Unit::setAgency(const std::string& _agency)
{
	agency = _agency;
}

void Unit::setStation(const std::string& _station)
{
	station = _station;
}

void Unit::setStatus(const unsigned int& _status, const std::string& _value)
{
	if (_status < MAX_STATUS) {
		status[_status] = _value;
	}
}

std::string Unit::getStatus(const unsigned int& _status)
{
	if (_status < MAX_STATUS) {
		return status[_status];
	}
	return std::string();
}

File: Oregon9-1-1/app/src/main/java/cad/oregon911/net/unit.java
package cad.oregon911.net;

/**
 * Created by Brandan on 12/28/2015.
 */
public class unit {

    public unit() {

    }

    public unit(int id, String name, String agency, String station, String[] status) {
        this.id = id;
        this.name = name;
        this.agency = agency;
        this.station = station;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAgency() {
        return agency;
    }

    public void setAgency(String agency) {
        this.agency = agency;
    }

    public String getStation() {
        return station;
    }

    public void setStation(String station) {
        this.station = station;
    }

    public String[] getStatus() {
        return status;
    }

    public void setStatus(String[] status) {
        this.status = status;
    }

    private int id;
    private String name;
    private String agency;
    private String station;
    private String[] status;
}
